### PR TITLE
Add GCR, Device Name, And Github Actions

### PR DESCRIPTION
**Pull request by maximillianbrain1**

### CODE REVIEW

File: pkg/scanner/scanner.go [hunk 1]
```diff
@@ -81,13 +82,37 @@ func (p *scanParser) checkPortsRemoved(host string) {
...
+}
 
 func (n *nmapWrapper) Run(ipAddresses []string, ctx context.Context) (result *nmap.Run, warnings []string, err error) {
+	if n.interfaceName != "" {
+		return n.runWithDevice(ipAddresses, ctx)
+	}
+	return n.runWithoutDevice(ipAddresses, ctx)
+}
+
+func (n *nmapWrapper) runWithDevice(ipAddresses []string, ctx context.Context) (result *nmap.Run, warnings []string, err error) {
```
Comment (cmparkinson) [inline pkg/scanner/scanner.go:96]: Tiny nit, but you can DRY this up a bit.  `nmap.NewScanner()` takes a slice of `func(*Scanner)` references as input.  You could build the slice in each of the `runXXXDevice` methods and then call another function that calls the `nmap.NewScanner()` function, handles the error and runs the scanner.  Something like: 

```golang
func (n *nmapWrapper) runWithDevice(ipAddresses []string, ctx context.Context) (result *nmap.Run, warnings []string, err error) {
  return runScanner(
    ipAddress,
    ctx,
    nmap.WithTargets(ipAddresses...),
    nmap.WithContext(ctx),
    nmap.WithSkipHostDiscovery(),
    nmap.WithInterface(n.interfaceName),
    nmap.WithUnprivileged(),
  )
}

func runScanner(ipAddresses []string, ctx context.Context, options ...func(*Scanner)) (result *nmap.Run, warnings []string, err error)
  scanner, err := nmap.NewScanner(options...)
  if err != nil {
    return nil, nil, fmt.Errorf("unable to create scanner scanner: %v", err)
  }
  return scanner.Run()
}
```

**Review by cmparkinson — APPROVED** [review]
LGTM.  One tiny style suggestion (that I'm not sure makes it any better).

File: pkg/scanner/scanner.go [hunk 2]
```diff
@@ -81,13 +82,37 @@ func (p *scanParser) checkPortsRemoved(host string) {
...
+	scanner, err := nmap.NewScanner(
+		nmap.WithTargets(ipAddresses...),
+		nmap.WithContext(ctx),
+		nmap.WithSkipHostDiscovery(),
+		nmap.WithInterface(n.interfaceName),
+		nmap.WithUnprivileged(),
+	)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to create scanner scanner: %v", err)
```
Comment (cmparkinson) [inline pkg/scanner/scanner.go:105]: You've also got an extra `scanner` in the log line.

File: pkg/scanner/scanner.go (hunk 1 above)
Comment (maximillianbrain1) [reply pkg/scanner/scanner.go]: Good call. I refactored it.

File: pkg/scanner/scanner.go (hunk 2 above)
Comment (maximillianbrain1) [reply pkg/scanner/scanner.go]: I fixed the typo with the refactor of the functions that had these log lines.

**Pull request merged**